ACKategories-iOS/Base/FlowCoordinator.swift

```diff
@@ -96,6 +97,15 @@ extension Base {
                 // VCs to remain in the navigation stack
                 let remainingViewControllers = Array(navigationController?.viewControllers[0..<index] ?? [])
                 navigationController?.setViewControllers(remainingViewControllers, animated: animated)
+
+                // set the appropriate value based on whether there are VCs remaining in the navigation stack
+                shouldCallDismissOnPresentingVC = remainingViewControllers.isEmpty
+            }
+
+            // ensure that dismiss will be called on presentingVC of root only when appropriate
+            if shouldCallDismissOnPresentingVC {
+                // dismiss when root was presented
+                rootViewController.presentingViewController?.dismiss(animated: animated)
```

Comment:
Does this look good? I'm a bit worried, that setting empty array using `navigationController?.setViewControllers(:animated:)` might be visible to the user...

Reply:
It should be OK, but better be safe than sorry. 🙂